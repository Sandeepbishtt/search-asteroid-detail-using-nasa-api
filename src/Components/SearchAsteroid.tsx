import React,{useEffect,useState} from 'react'
import {Box,Input,Button,CircularProgress,Alert,AlertTitle} from '@mui/material'
import {useDispatch,useSelector} from 'react-redux'
import {useHistory} from 'react-router-dom'
import axios from 'axios'
import {addData,fetchData} from '../Redux/Reducer'

const classes = {
spinner:{
	marginLeft:'50%'
},
form:{
	margin:'2rem'
},
button:{
	marginLeft:'5rem'
}
}




const SearchAsteroid :React.FC= () =>{
const dispatch= useDispatch()
const data = useSelector(fetchData)
const history =useHistory()
const [showSpinner,setShowSpinner] = useState(true)


useEffect(()=>{
	axios.get(`https://api.nasa.gov/neo/rest/v1/neo/browse?api_key=CBH09wOsGOJRJs55CYRSTc8jaDQhSMY22ImnKytP`)
.then((response) =>{
	dispatch(addData(response.data))
	setShowSpinner(false)
})
.catch(error => console.log(error))
},[])

const [input,setInput] = useState('')
const [showAlert,setShowAlert] = useState(false)
const submitHandler = (e:React.FormEvent) =>{
	e.preventDefault()
	data.near_earth_objects.forEach((val:any,index:number) =>{
		if(val.id===input){
			history.push({
				pathname:'/AsteroidDetail',
				state:{val}
			})
		}
		else{
			setShowAlert(true)
		}
	})
	setInput('')
}

useEffect(()=>{
setTimeout(()=>{
	setShowAlert(false)
},3000)
},[showAlert])


const randomIdHandler = (e:React.MouseEvent) =>{
const idArray = data.near_earth_objects
const id = idArray[Math.floor(Math.random()*idArray.length)].id
idArray.forEach((val:any,index:number) =>{
		if(val.id===id){
			history.push({
				pathname:'/AsteroidDetail',
				state:{val}
			})
		}
	})
}





	return(
<>
{showAlert && 
<Alert severity='error'>
<AlertTitle>Error</AlertTitle>
This is an error Alert -<strong> Asteroid ID not found </strong>
</Alert>
}
{showSpinner ? <CircularProgress size={80} style={classes.spinner}   /> :
<Box
border={1}
borderColor='grey.500'
bgcolor='white.main'
borderRadius={5}
boxShadow={3}
width={400}
height={200}
mx="auto"
mt='2rem'
data-test = 'box'
>
<form onSubmit={submitHandler} style={classes.form}>
<Input
type='text'
required
className='form-control'
placeholder= 'Enter Asteroid ID'
onChange={(e:React.ChangeEvent<HTMLInputElement>) => setInput(e.target.value)}
value={input}
/>
<Button
type='submit'
variant='contained'
color='primary'
disabled={input===""}

>
Submit</Button>
</form>
<Button
type='submit'
variant='contained'
color='primary'
style={classes.button}
onClick={randomIdHandler}
>
Random Asteroid</Button>
</Box>
}
</>

		)
}

export default SearchAsteroid