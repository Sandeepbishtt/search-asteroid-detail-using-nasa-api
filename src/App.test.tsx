import React from 'react';
import { render, screen } from '@testing-library/react';
import {BrowserRouter} from 'react-router-dom'
import {Provider} from 'react-redux'
import store from './Redux/Store'
import SearchAsteroid from './Components/SearchAsteroid';
import "@testing-library/jest-dom";
test('renders learn react link', () => {
  render( <React.StrictMode>
    <BrowserRouter>
      <Provider store={store}>
        <SearchAsteroid/>
      </Provider>
    </BrowserRouter>
  </React.StrictMode>,);
  const linkElement = screen.getByText(/Search Asteroid/i);
  expect(linkElement).toBeInTheDocument();
});