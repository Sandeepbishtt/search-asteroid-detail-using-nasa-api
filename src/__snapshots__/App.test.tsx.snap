// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check snapshot render snapshot 1`] = `
<div>
  <h2
    style="text-align: center;"
  >
     Nasa Asteroid Detail App 
  </h2>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style="width: 80px; height: 80px; margin-left: 50%;"
  >
    <svg
      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</div>
`;
